Source/ARTDefault.m

```diff
@@ -6,8 +6,37 @@
 //  Copyright (c) 2015 Ably. All rights reserved.
 //
 
+#import "Ably.h"
 #import "ARTDefault+Private.h"
 #import "ARTNSArray+ARTFunctional.h"
+#import <sys/utsname.h>
+
+// NSOperatingSystemVersion has NSInteger as version components for some reason, so mitigate it here.
+static inline UInt32 conformVersionComponent(const NSInteger component) {
+    return (component < 0) ? 0 : (UInt32)component;
+}
+
+NSString *canonizeStringAsAgentToken(NSString *const inputString) {
```

Comment:
Thanks for all your work on defining this method, but with our current requirements for `Ably-Agent` it is superfluous to requirements. Please remove it as any code emitting product names to the agents string must be capable of being statically validated at compile time (i.e. product names must not be dynamically sourced or otherwise generated at runtime).

Reply:
https://github.com/ably/ably-cocoa/pull/1119/commits/088bec2fc18adf12ef8528fd2f39f3da758c005d